packages/sdk/src/types/runtime/dialog.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { EventEmitter } from 'events';
+
+import { Context, GroupMask, Texture, User } from '.';
+
+export enum DialogIcon {
+	/** A speech bubble */
+	Info = 'info',
+	/** A yellow exclamation point */
+	Warning = 'warning',
+	/** A red exclamation point */
+	Error = 'error',
+	/** A question mark */
+	Question = 'question'
+}
+
+export interface DialogResponse {
+	user: User;
+	button: string;
+	text?: string;
+}
+
+/** Pop a message to users, and get a response */
+export class Dialog extends EventEmitter {
+	/**
+	 * Create a new dialog box to present to users.
+	 * @param context An active MRE context.
+	 * @param options.text The message to present.
+	 * @param options.buttons Label strings for dialog buttons. The user response will contain this string.
+	 * @param options.input Whether the dialog should present a text input. Defaults to false.
+	 * @param options.recipients Who to send the message to. Defaults to all users.
+	 * @param options.icon A graphic to display next to the text. Defaults to none.
+	 * @param options.persistent Send this message to late-joining users. Defaults to false.
+	 */
+	public constructor(private context: Context, options: {
+		text: string,
+		buttons: string[],
```

Comment:
Would we enforce a maximum of two button labels?

Reply:
I can think of three-button use cases (retry, ignore, cancel). Are there use cases that need more, say 10? If it's easy for us to reflow the button panel, I see no need to arbitrarily limit it.